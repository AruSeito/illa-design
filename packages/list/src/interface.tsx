import { HTMLAttributes } from "react"

export interface ListProps extends HTMLAttributes<HTMLDivElement> {}
