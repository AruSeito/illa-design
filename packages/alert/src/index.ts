export * from "./alert"
export * from "./interface"
