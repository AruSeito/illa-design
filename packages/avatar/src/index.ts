export * from "./avatar"
export * from "./interface"
export * from "./avatar-group"