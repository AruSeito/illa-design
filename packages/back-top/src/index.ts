export * from "./backtop"
export * from "./interface"
export * from "./style"
