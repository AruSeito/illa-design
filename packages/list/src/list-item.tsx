import { forwardRef } from "react"
import { ListItemProps } from "./interface"
import {
  applyListItemActionsStyle,
  applyListItemContainer,
  applyListItemExtraStyle,
  applyListItemInner,
} from "./style"
import { cx } from "@emotion/css"

export const ListItem = forwardRef<HTMLDivElement, ListItemProps>(
  (props, ref) => {
    const { actions, extra, className, ...otherProps } = props
    return (
      <div
        className={cx(applyListItemContainer, className)}
        ref={ref}
        {...otherProps}
      >
        <div className={applyListItemInner}>
          {props.children}
          {actions && (
            <div className={applyListItemActionsStyle}>{actions}</div>
          )}
        </div>
        {extra && <div className={applyListItemExtraStyle}>{extra}</div>}
      </div>
    )
  },
)

ListItem.displayName = "ListItem"
