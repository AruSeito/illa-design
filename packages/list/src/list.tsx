/** @jsxImportSource @emotion/react */
import { forwardRef } from "react"
import { ListProps } from "."

export const List = forwardRef<HTMLDivElement, ListProps>((props, ref) => {
  const { ...otherProps } = props
  return (
    <div ref={ref} {...otherProps}>
      {props.children}
    </div>
  )
})

List.displayName = "List"
