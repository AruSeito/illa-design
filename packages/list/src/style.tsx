import { css } from "@emotion/css"
import { globalColor, illaPrefix } from "@illa-design/theme"
import { ListSize } from "./interface"

export function applyListContainer(bordered?: boolean): string {
  let containerCss = css`
    display: inline-flex;
    min-width: 686px;
    flex-direction: column;
  `
  if (bordered) {
    containerCss = css`
      ${containerCss};
      border: solid 1px ${globalColor(`--${illaPrefix}-gray-09`)};
      border-radius: 2px;
    `
  }
  return containerCss
}

export function applyListItemOuter(
  size: ListSize,
  hoverable?: boolean,
): string {
  let sizeCss: string
  switch (size) {
    case "small":
      sizeCss = css`
        padding: 8px 24px;
      `
      break
    case "medium":
      sizeCss = css`
        padding: 12px 24px;
      `
      break
    case "large":
      sizeCss = css`
        padding: 16px 24px;
      `
      break
  }
  let hoverableCss: string = css``
  if (hoverable) {
    hoverableCss = css`
      &:hover {
        background-color: ${globalColor(`--${illaPrefix}-gray-09`)};
      }
    `
  }
  return css`
    ${sizeCss};
    ${hoverableCss};
  `
}

export const applyListItemContainer = css`
  display: flex;
  flex-direction: row;
`

export const applyListItemInner = css`
  display: inline-flex;
  flex-direction: column;
  flex-grow: 1;
`

export const applyItemMetaContainer = css`
  display: inline-flex;
  align-items: start;
  flex-direction: row;
`

export const applyItemMetaInner = css`
  display: inline-flex;
  flex-direction: row;
`

export const applyItemMetaAvatar = css`
  margin-right: 8px;
`

export const applyTypoStyle = css`
  align-self: center;
`

export const applyItemMetaTitle = css`
  margin-bottom: 4px;
`

export const applyListItemActionsStyle = css`
  display: inline-flex;
  margin-top: 8px;
`

export const applyListItemExtraStyle = css`
  display: inline-flex;
  margin-left: 32px;
`

export function applyBarStyle(size: ListSize) {
  let sizeCss: string
  switch (size) {
    case "small":
      sizeCss = css`
        padding: 8px 24px;
      `
      break
    case "medium":
      sizeCss = css`
        padding: 12px 24px;
      `
      break
    case "large":
      sizeCss = css`
        padding: 16px 24px;
      `
      break
  }
  return css`
    ${sizeCss};
  `
}
