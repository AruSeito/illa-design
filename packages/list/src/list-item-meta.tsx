import { forwardRef } from "react"
import { ListItemMetaProps } from "./interface"
import {
  applyItemMetaAvatar,
  applyItemMetaContainer,
  applyTypoStyle,
} from "./style"
import { Avatar } from "@illa-design/avatar"
import { Paragraph, Text, Typography } from "@illa-design/typography"
import { cx, globalColor, illaPrefix } from "@illa-design/theme"

export const ListItemMeta = forwardRef<HTMLDivElement, ListItemMetaProps>(
  (props, ref) => {
    const { title, avatar, description, className, ...otherProps } = props
    return (
      <div
        ref={ref}
        className={cx(applyItemMetaContainer, className)}
        {...otherProps}
      >
        {avatar && (
          <Avatar
            className={applyItemMetaAvatar}
            size="medium"
            shape="square"
            src={avatar}
          />
        )}
        <Typography className={applyTypoStyle}>
          {title && (
            <Text fontSize="14px" colorScheme="gray" bold>
              {title}
            </Text>
          )}
          {title && description && <div style={{ height: "4px" }} />}
          {description && (
            <Paragraph
              colorScheme={
                title == undefined
                  ? globalColor(`--${illaPrefix}-gray-02`)
                  : globalColor(`--${illaPrefix}-gray-04`)
              }
              fontSize="14px"
            >
              {description}
            </Paragraph>
          )}
        </Typography>
      </div>
    )
  },
)

ListItemMeta.displayName = "ListItemMeta"
