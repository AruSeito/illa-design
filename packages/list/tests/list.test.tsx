import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { List } from "../src"

test("List renders correctly", () => {
  render(<List data-testid="test-list">List</List>)
  expect(screen.getByTestId("test-list")).toBeInTheDocument()
})
