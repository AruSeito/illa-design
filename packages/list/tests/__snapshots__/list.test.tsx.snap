// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders without border and split 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 686px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: solid 1px #f5f5f5;
  border-radius: 2px;
}

.emotion-1 {
  padding: 12px 24px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-5 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-6 {
  word-break: break-all;
}

.emotion-6 h1 {
  margin: 0;
  font-size: 36px;
  line-height: normal;
}

.emotion-6 h2 {
  margin: 0;
  font-size: 32px;
  line-height: normal;
}

.emotion-6 h3 {
  margin: 0;
  font-size: 28px;
  line-height: normal;
}

.emotion-6 h4 {
  margin: 0;
  font-size: 24px;
  line-height: normal;
}

.emotion-6 h5 {
  margin: 0;
  font-size: 20px;
  line-height: normal;
}

.emotion-6 h6 {
  margin: 0;
  font-size: 16px;
  line-height: normal;
}

.emotion-6 p {
  margin: 0;
}

.emotion-7 {
  margin-bottom: 4px;
}

.emotion-8 {
  font-size: 14px;
}

.emotion-9 {
  color: rgba(0, 0, 0, 0.64);
  font-weight: bold;
}

.emotion-10 {
  font-size: 14px;
}

.emotion-11 {
  color: #999;
}

.emotion-12 {
  display: inline-flex;
  vertical-align: middle;
  border-color: #ebebeb;
  border-style: solid;
  border-width: 0 0 1px 0;
  width: 100%;
}

<div
  class="emotion-0"
  data-testid="test-list"
>
  <div
    class="rc-virtual-list"
    style="position: relative;"
  >
    <div
      class="rc-virtual-list-holder"
    >
      <div>
        <div
          class="rc-virtual-list-holder-inner"
          style="display: flex; flex-direction: column;"
        >
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-4"
                >
                  <article
                    class="emotion-5 emotion-6"
                  >
                    <span
                      class="emotion-7 emotion-8"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-9"
                        >
                          Title A
                        </span>
                      </span>
                    </span>
                    <div
                      class="emotion-10"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-11"
                        >
                          Desc A
                        </span>
                      </span>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
          <div
            class="emotion-12"
          />
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <div
                  class="emotion-4"
                >
                  <article
                    class="emotion-5 emotion-6"
                  >
                    <span
                      class="emotion-7 emotion-8"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-9"
                        >
                          Title A
                        </span>
                      </span>
                    </span>
                    <div
                      class="emotion-10"
                    >
                      <span
                        style="position: relative;"
                      >
                        <iframe
                          style="background: transparent; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%; z-index: -1;"
                        />
                        <span
                          class="emotion-11"
                        >
                          Desc A
                        </span>
                      </span>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
